crates/holochain/src/conductor/conductor.rs

```diff
@@ -867,31 +871,99 @@ mod network_impls {
-                ConductorApiError::other(format!("App not installed: {}", installed_app_id))
-            })?;
+            let mut app_dnas = self.get_dna_hashes_for_app(installed_app_id).await?;
 
-            let mut app_dnas: HashSet<DnaHash> = HashSet::new();
-            for cell_infos in app_info.cell_info.values() {
-                for cell_info in cell_infos {
-                    let dna = match cell_info {
-                        CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
-                        CellInfo::Cloned(cell) => cell.cell_id.dna_hash().clone(),
-                        CellInfo::Stem(cell) => cell.original_dna_hash.clone(),
-                    };
-                    app_dnas.insert(dna);
-                }
+            if let Some(dna_hashes) = maybe_dna_hashes {
+                app_dnas.retain(|h| dna_hashes.contains(h));
+            };
+
+            self.get_agent_infos(Some(app_dnas)).await
+        }
+
+        /// Get the content of the peer meta store(s) for an agent at a given Url
+        /// for spaces (dna hashes) of a specific app
+        pub async fn app_agent_meta_info(
+            &self,
+            installed_app_id: &InstalledAppId,
+            url: Url,
+            maybe_dna_hashes: Option<Vec<DnaHash>>,
+        ) -> ConductorApiResult<BTreeMap<DnaHash, Vec<AgentMetaInfo>>> {
+            let mut app_hashes = self.get_dna_hashes_for_app(installed_app_id).await?;
+            if let Some(dna_hashes) = maybe_dna_hashes {
+                app_hashes.retain(|h| dna_hashes.contains(h));
             }
+            self.agent_meta_info(url, Some(app_hashes)).await
+        }
 
-            let hashes = match maybe_dna_hashes {
-                Some(mut dna_hashes) => {
-                    dna_hashes.retain(|h| app_dnas.contains(h));
-                    dna_hashes
-                }
-                None => app_dnas.into_iter().collect(),
-            };
-            self.get_agent_infos(Some(hashes)).await
+        /// Get the content of the peer meta store(s) for an agent at a given Url
+        pub async fn agent_meta_info(
+            &self,
+            url: Url,
+            maybe_dna_hashes: Option<Vec<DnaHash>>,
+        ) -> ConductorApiResult<BTreeMap<DnaHash, Vec<AgentMetaInfo>>> {
+            let mut space_ids = self
+                .spaces
+                .get_from_spaces(|space| (*space.dna_hash).clone());
+
+            if let Some(dna_hashes) = maybe_dna_hashes {
+                space_ids.retain(|dna_hash| dna_hashes.contains(dna_hash));
+            }
+
+            if space_ids.is_empty() {
+                return Err(ConductorApiError::Other(
+                    "No cell found for the provided dna hashes.".into(),
+                ));
+            }
+
+            // Sort to get consistently ordered BTreeMap
+            space_ids.sort();
+
+            let mut all_infos = BTreeMap::new();
```

Comment:
Had to change to `BtreeMap` here to be able to test it in `hc sandbox`, it wouldn't give out consistently ordered output on stdout.